steampipeconfig/connection_data.go

```diff
@@ -28,6 +34,43 @@ type ConnectionData struct {
 	ModTime time.Time
 }
 
+// ConnectionData is a struct containing all details for a connection
+// - the plugin name and checksum, the connection config and options
+// json tags needed as this is stored in the connection state file
+type ConnectionData struct {
+	StructVersion int64 `json:"struct_version,omitempty"`
+	// the fully qualified name of the plugin
+	Plugin string `json:"plugin,omitempty"`
+	// the underlying connection object
+	Connection *modconfig.Connection `json:"connection,omitempty"`
+	// schema mode - static or dynamic
+	SchemaMode string `json:"schema_mode,omitempty"`
+	// the hash of the connection schema
+	SchemaHash string `json:"schema_hash,omitempty"`
+	// the creation time of the plugin file (only used for local plugins)
+	ModTime time.Time `json:"mod_time"`
+}
+
+func (s ConnectionData) IsValid() bool {
+	return s.StructVersion > 0
+}
+
+func (s *ConnectionData) MigrateFrom(legacyState interface{}) migrate.Migrateable {
+	old := legacyState.(LegacyConnectionData)
+	s.StructVersion = constants.StructVersion
```

Comment:
we already have a value defined for this struct version
`s.StructVersion = ConnectionDataStructVersion`